lasagne/layers/conv.py

```diff
@@ -37,6 +86,97 @@ def conv_output_length(input_length, filter_size,
 
 
 class Conv1DLayer(Layer):
+    """
+    1D convolutional layer
+
+    Performs a 1D convolution on its input and optionally adds a bias and
+    applies an elementwise nonlinearity.
+
+    Parameters
+    ----------
+    incoming : a :class:`Layer` instance or a tuple
+        The layer feeding into this layer, or the expected input shape. The
+        output of this layer should be a 3D tensor, with shape
+        `(batch_size, num_input_channels, input_length)`.
+
+    num_filters : int
+        The number of learnable convolutional filters this layer has.
+
+    filter_size : int or tuple of int
+        An integer or a 1-element tuple specifying the size of the filters.
+
+    stride : int or tuple of int
+        An integer or a 1-element tuple specifying the stride of the
+        convolution operation.
+
+    border_mode : str, one of 'valid', 'full', 'same'
+        A string indicating the convolution border mode.
+
+        If 'valid', the convolution is only computed where the input and the
+        filter fully overlap.
+
+        If 'full', the convolution is computed wherever the input and the
+        filter overlap by at least one position.
+
+        If 'same', the convolution is computed wherever the input and the
+        filter overlap by half, which results in an output length that is the
+        same as the input length.
+
+    untie_biases : bool, default False
+        If False, the layer will have a bias parameter for channel, which is
+        shared across all positions in this channel. As a result, the `b`
+        attribute will be a vector (1D).
+
+        If True, the layer will have separate bias parameters for each
+        position in each channel. As a result, the `b` attribute will be a
+        matrix (2D).
+
+    W : Theano shared variable, numpy array or callable
+        An initializer for the weights of the layer. If a Theano shared
+        variable is provided, it is used unchanged. If a numpy array is
+        provided, a shared variable is created and initialized with the
+        array. If a callable is provided, a shared variable is created and
+        the callable is called with the desired shape to generate suitable
+        initial values. The variable is then initialized with those values.
+
+        This should initialize the layer weights to a 3D array with shape
+        `(num_filters, num_input_channels, filter_length)`.
+
+    b : Theano shared variable, numpy array, callable or None
+        An initializer for the biases of the layer. If a Theano shared
+        variable is provided, it is used unchanged. If a numpy array is
+        provided, a shared variable is created and initialized with the
+        array. If a callable is provided, a shared variable is created and
+        the callable is called with the desired shape to generate suitable
+        initial values. The variable is then initialized with those values.
```

Comment:
Very good, but I wonder whether we really want to copy that text for each and every parameter of each and every layer. `b` could point to `W`, or we could have that description in some central place (I don't know where, though).